src/outmdsd/DataReader.cc

```diff
@@ -87,6 +87,7 @@ DataReader::DoRead(
     }
     catch(const SocketException & ex) {
         Log(TraceLevel::Info, "SocketException " << ex.what());
+        return false;
```

Comment:
When a SocketException occurs, DoRead() shouldn't return false. Here is why: this plugin is designed to be socket-failure tolerant. When there is socket error while talking to mdsd, the plugin would keep on do periodically retry until it succeeds. Meanwhile, the event data will be queued/persisted in fluentd process to avoid data loss.

If there is partial data sent to mdsd during socket exception, mdsd won't do ack. if there is no ack, the plugin will resend the same msg before the msg is expired.

the plugin parameter "acktimeoutms" is used to control this ack timeout.